doc/user/modes.md

```diff
@@ -0,0 +1,27 @@
+Fact Modes
+==========
+
+Fact modes only affect input. Compute semantics are the same regardless of the fact mode for the current attribute.
+
+The following fact modes are supported:
+
+1. Event
+2. Sparse States
+3. Dense States
+
+Event
+-----
+
+This is the default fact mode. Each fact is ingested as a single value.
+
+Sparse States
+-------------
+
+Facts are treated as state updates. A running state is kept and updated with each new fact.
+
+### Notes
+
+1. Number of facts: one extra fact with the current fact state is inserted with the snapshot/chord date.
```

Comment:
I actually think it needs to keep its original date as well (or only the original date).